File: leetcode/top_interview_150/array_string/27. Remove Element/main.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) < 1:
            return -1
        left, right = 0, len(nums) - 1
        while left <= right:
            if nums[right] == val:
                right -= 1
                continue
            if nums[left] == val:
                nums[left], nums[right] = nums[right], nums[left]
            left += 1
        return left


if __name__ == "__main__":
    # print(Solution().)
    pass

File: leetcode/top_interview_150/array_string/27. Remove Element/main.go
package main

import "fmt"

func removeElement(nums []int, val int) int {
	if len(nums) < 1 {
		return -1
	}
	var (
		left  = 0
		right = len(nums) - 1
	)

	for left <= right {
		if nums[right] == val {
			right--
			continue
		}
		if nums[left] == val {
			nums[left], nums[right] = nums[right], nums[left]
		}
		left++
	}
	return left
}

func main() {
	fmt.Println(removeElement([]int{3, 2, 2, 3}, 3))
	fmt.Println(removeElement([]int{0, 1, 2, 2, 3, 0, 4, 2}, 2))
}
